AsyncDisplayKit/ASTextNode.mm

```diff
@@ -76,7 +71,7 @@ @implementation ASTextNode {
   NSRange _highlightRange;
   ASHighlightOverlayLayer *_activeHighlightLayer;
 
-  ASDN::Mutex _rendererLock;
+  std::recursive_mutex _textLock;
```

Comment:
We decided to start with a recursive mutex for now. As I wrote in the commit message we need to / will rethink the whole locking strategy starting from `ASDisplayNode` pretty soon and with that I would like to remove our recursive mutexes. Unfortunately this is not possible at the moment in terms of time :/